src/EntityFramework.Core/Metadata/Conventions/Internal/PropertyMappingConvention.cs

```
@@ -0,0 +1,87 @@
+                }
+
+                if (entityType.HasClrType)
+                {
+                    var clrProperties = new HashSet<string>();
+                    clrProperties.UnionWith(entityType.ClrType.GetRuntimeProperties().Where(pi => pi.IsCandidateProperty()).Select(pi => pi.Name));
+
+                    clrProperties.ExceptWith(entityType.Properties.Select(p => p.Name));
+
+                    clrProperties.ExceptWith(entityType.Navigations.Select(p => p.Name));
+
+                    var entityTypeBuilder = modelBuilder.Entity(entityType.ClrType, ConfigurationSource.Convention);
+
+                    clrProperties.RemoveWhere(p => entityTypeBuilder.IsIgnored(p, ConfigurationSource.Convention));
+
+                    if (clrProperties.Count > 0)
+                    {
+                        foreach (var clrProperty in clrProperties)
+                        {
+                            var actualProperty = entityType.ClrType.GetRuntimeProperties().First(pi => pi.Name == clrProperty);
+                            var targetType = FindCandidateNavigationPropertyType(actualProperty);
+                            if (targetType != null)
+                            {
+                                if (!modelBuilder.IsIgnored(targetType.FullName, ConfigurationSource.Convention))
+                                {
+                                    throw new InvalidOperationException(CoreStrings.NavigationNotAdded(actualProperty.Name, entityType.Name));
+                                }
+                            }
+                            else
+                            {
+                                targetType = actualProperty.PropertyType;
+                                targetType = targetType.TryGetSequenceType() ?? targetType;
+                                targetType = targetType.UnwrapNullableType();
+                                if (!targetType.GetTypeInfo().IsInterface)
```

Sorry, I haven't looked at the code closely, just seeing the comments. When you say "we would not add them to the model" does this mean we throw indicating we don't know how to map them? I think the only thing we should silently ignore (not map) based on our current thinking would be read-only non-navigation properties, since these are likely to be computed properties and we think in the majority of cases the application would want them not mapped. (Although it could be argued that we also throw for these.) For every other property that we can't map we should throw unless it is explicitly ignored.